swift/ql/test/library-tests/dataflow/taint/url.swift

```diff
@@ -61,7 +61,8 @@ func taintThroughURL() {
 	sink(arg: urlClean)
 	sink(arg: urlTainted) // $ tainted=57
 	sink(arg: urlTainted.absoluteURL) // $ tainted=57
-	sink(arg: urlTainted.baseURL) // $ tainted=57
+	sink(arg: urlTainted.baseURL) // $ Safe
```

Comment:
Ah that's interesting, so if I'm understanding correctly `URL("http://example.com/abc")` and `URL("abc", URL("http://example.com/"))` are not two ways of constructing exactly the same thing - because the latter has a `baseURL` (amongst other visible differences).

Reply:
Correct. It seems that if the first argument is a relative path it only overrides the path, query, and fragment parts of the `relativeTo` URL, but then the original is saved as `baseURL`. With the one-argument constructor `baseURL` is apparently never populated.